master/internal/api_project.go

```diff
@@ -1076,3 +1076,31 @@ func (a *apiServer) GetProjectsByUserActivity(
 
 	return &apiv1.GetProjectsByUserActivityResponse{Projects: viewableProjects}, nil
 }
+
+func (a *apiServer) GetMetadataValues(
+	ctx context.Context, req *apiv1.GetMetadataValuesRequest,
+) (*apiv1.GetMetadataValuesResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	p, err := a.GetProjectByID(ctx, req.ProjectId, *curUser)
+	if err != nil { // Can view project?
+		return nil, err
+	}
+
+	values := []string{}
+	err = db.Bun().NewSelect().Distinct().
+		Table("runs_metadata_index").
+		Column("string_value").
+		Where("string_value IS NOT NULL").
+		Where("project_id=?", p.Id).
+		Where("flat_key=?", req.Key).Scan(ctx, &values)
+	if err != nil {
```

Comment:
Are we wanting to return an error here if the above select query comes back empty or an empty response?

Reply:
I think an empty response would be better here. If there aren't any values for this key in the project then frontend can just not show any dropdown.

Reply:
Sounds good, doesn't the query return `sql.ErrNoRows` if the response is empty?

Reply:
No. I added a test for this case